addons/mail/views/assets.xml

```diff
@@ -217,6 +217,7 @@
 
         <template id="assets_tests" name="Mail Assets Tests" inherit_id="web.assets_tests">
             <xpath expr="." position="inside">
+                <script type="text/javascript" src="/web/static/tests/helpers/test_utils_file.js"/>
```

Comment:
shouldn't this be in `web` bundle?

Reply:
@seb-odoo ?

Reply:
I don't really care, but it's not used in `web` module.

Someone just installing and running `web` tours (without `mail` module) does not need to load this file.
I think it should be moved to `web` only the day someone else needs it in a module that does not depend on `mail`?

But if we think it's better to add it preemptively on `web` I really don't mind.

cc @aab-odoo maybe you can decide for us? And maybe you'd also want to check the small test util file function that we moved from `mail` into `web`.

Reply:
I don't care either. The file is small, it's fine to put it in the bundle in web. But the function `_createFakeDataTransfer` it contains is incorrect: if the `getData` function it returns is called, it will crash.

Reply:
I didn't touch that function and I don't use it in `inputFiles` (we need real dataTransfer as we need a FileList object to allow assignation to file input) so I don't know what do you expect for that ?

Reply:
@alexkuhn has the authorship, as he was in this thread I simply dropped the information. Maybe could he fix it?

Reply:
I think `_createFakeDataTransfer` is necessary when business code wants to have custom `dropEffect`, like have `'copy'` on drag over. Problem with `new DataTransfer()` is that it defaults to `{ dropEffect: 'none', effectAllowed: 'none' }` and we can't change to `{ dropEffect: 'all', effectAllowed: 'all' }`.

Other than that, I think using `const dt = new DataTransfer()` and `dt.items.add(...files)` is much better than using the fake data transfer, especially with the requirement of `files` to be a `FileList` on input (`FileList` has no constructor, so likely a browser security measure).

Reply:
I don't know about all this, I was just talking about the typo (I guess) in getData: `file` -> `files`

Reply:
> I don't know about all this, I was just talking about the typo (I guess) in getData: `file` -> `files`

Indeed a typo, some test used `.getData` of fake DT before, but no longer the case since the change.

Instead of fixing the typo, I would just remove the fake data transfer and just use a genuine data transfer.

The problem of `dropEffect` is the only one I can think of, and it's basically minor so I wouldn't mind removing assertion on `dropEffect` in existing tests and consider this small feature not testable. Small price to pay for more robust tests on files.

Reply:
I fixed the typo just to be sure, didn't do all the rest as I think it's not in the scope of this PR.
I don't know if I should create a task in discuss or web for all the changes @alexkuhn is mentioning ?

Reply:
FYI task-2314954

Reply:
:+1: